adapter-cloudflare: allow disabling generation of fallback `404.html` (#9762)
sveltejs/kit: documentation/docs/25-build-and-deploy/60-adapter-cloudflare.md

You can have up to 100 `include` and `exclude` rules combined. Generally you can omit the `routes` options, but if (for example) your `<prerendered>` paths exceed that limit, you may find it helpful to manually create an `exclude` list that includes `'/articles/*'` instead of the auto-generated `['/articles/foo', '/articles/bar', '/articles/baz', ...]`.

### Generate 404

The `generate404` option allows you to specify whether a fallback `404.html` page will be generated.

> The default value of `auto` will only generate a fallback `404.html` page if there are *any* prerendered pages.
> 
> You may want to set this to `false` if you have a fully dynamic app that does not prerender pages during build time.

Reviewer: ```suggestion
The default value of `auto` or `true` will generate a fallback page if prerendering occurs. This will be generated from your [error page](https://kit.svelte.dev/docs/routing#error).
```

> You may want to set this to `false` if you have a fully dynamic app that does not prerender pages during build time.

I don't think we need to say this, as `auto` would prevent this anyway? If we want to mention `false` then we could say something like `Set this option to `false` to prevent generating a fallback.`

You can have up to 100 `include` and `exclude` rules combined. Generally you can omit the `routes` options, but if (for example) your `<prerendered>` paths exceed that limit, you may find it helpful to manually create an `exclude` list that includes `'/articles/*'` instead of the auto-generated `['/articles/foo', '/articles/bar', '/articles/baz', ...]`.

### Generate 404

The `generate404` option allows you to specify whether a fallback `404.html` page will be generated.

The default value of `auto` will only generate a fallback `404.html` page if there are *any* prerendered pages.

You may want to set this to `false` if you have a fully dynamic app that does not prerender pages during build time.

> Otherwise, it will default to `true`.

Reviewer 2: ```suggestion
```

You can have up to 100 `include` and `exclude` rules combined. Generally you can omit the `routes` options, but if (for example) your `<prerendered>` paths exceed that limit, you may find it helpful to manually create an `exclude` list that includes `'/articles/*'` instead of the auto-generated `['/articles/foo', '/articles/bar', '/articles/baz', ...]`.

### Generate 404

> The `generate404` option allows you to specify whether a fallback `404.html` page will be generated.

Reviewer 2: I wonder if we should rename this to `generateFallback` so it better matches with the error message users get during build.
```bash
> Using @sveltejs/adapter-cloudflare
file:///Users/my-app/node_modules/@sveltejs/kit/src/core/postbuild/fallback.js:53
        throw new Error(`Could not create a fallback page — failed with status ${response.status}`);
```

The default value of `auto` will only generate a fallback `404.html` page if there are *any* prerendered pages.

> You may want to disable `404.html` if you have a fully dynamic app that should not process pages during build time.

Reviewer 2: ```suggestion
You may want to set this to `false` if you have a fully dynamic app that does not prerender pages during build time.
```
